packages/flutter/lib/src/services/text_input.dart

```diff
@@ -2609,27 +2675,264 @@ class SystemContextMenuController with SystemContextMenuClient {
   ///    the system context menu is supported on the current platform.
   Future<void> hide() async {
     assert(!_isDisposed);
-    // This check prevents a given instance from accidentally hiding some other
+    // This check prevents the instance from accidentally hiding some other
     // instance, since only one can be visible at a time.
     if (this != _lastShown) {
       return;
     }
     _lastShown = null;
+    _buttonCallbacks.clear();
+    ServicesBinding.systemContextMenuClient = null;
     // This may be called unnecessarily in the case where the user has already
     // hidden the menu (for example by tapping the screen).
     return _channel.invokeMethod<void>('ContextMenu.hideSystemContextMenu');
   }
 
   @override
   String toString() {
-    return 'SystemContextMenuController(onSystemHide=$onSystemHide, _hiddenBySystem=$_hiddenBySystem, _isVisible=$_isVisible, _isDisposed=$_isDisposed)';
+    return 'SystemContextMenuController(onSystemHide=$onSystemHide, _hiddenBySystem=$_hiddenBySystem, _isVisible=$isVisible, _isDisposed=$_isDisposed)';
   }
 
   /// Used to release resources when this instance will never be used again.
   void dispose() {
     assert(!_isDisposed);
     hide();
-    ServicesBinding.unregisterSystemContextMenuClient(this);
     _isDisposed = true;
   }
 }
+
+/// Describes a context menu button that will be rendered in the system context
+/// menu.
+///
+/// See also:
+///
+///  * [SystemContextMenuController], which is used to show the system context
+///    menu.
+///  * [IOSSystemContextMenuItem], which performs a similar role but at the widget
+///    level, where the titles can be replaced with default localized values.
+///  * [ContextMenuButtonItem], which performs a similar role for Flutter-drawn
+///    context menus.
+@immutable
+sealed class IOSSystemContextMenuItemData {
+  const IOSSystemContextMenuItemData();
+
+  /// The callback to be called when the menu item is pressed.
+  ///
+  /// Not exposed for built-in menu items, which handle their own action when
+  /// pressed.
+  VoidCallback? get onPressed => null;
+
+  /// The text to display to the user.
+  ///
+  /// Not exposed for some built-in menu items whose title is always set by the
```

Comment:
Would it also be beneficial to have a list of these items?

Reply:
Covered in another comment.